connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -763,6 +789,30 @@ private boolean startTask(ConnectorTaskId taskId) {
         );
     }
 
+    private Callable<Void> getTaskCallable(final ConnectorTaskId taskId, final TaskStatus.State targetState) {
+        return new Callable<Void>() {
+            @Override
+            public Void call() throws Exception {
+                switch (targetState) {
+                    case RUNNING:
+                        try {
+                            startTask(taskId);
```

Comment:
Upon checking out the code, actually the only purpose is for the running of the tasks, and not starting at all :-)

It could definitely do with a better name... and naming for the the threads with a `ThreadFactory` (we should do that for the `bulkExecutor` too)